crates/api_models/src/payments.rs

```diff
@@ -775,6 +778,13 @@ pub struct CryptoData {
     pub pay_currency: Option<String>,
 }
 
+#[derive(Debug, Clone, Eq, PartialEq, serde::Deserialize, serde::Serialize, ToSchema)]
+#[serde(rename_all = "snake_case")]
+pub struct UpiData {
+    #[schema(value_type = Option<String>, example = "successtest@iata")]
+    pub vpa_id: Option<Secret<String>>,
```

Comment:
We'll need a masking strategy specific for UPI VPAs. I'll open an issue for the same.

Reply:
I've opened #1578 for this, resolving this conversation.